app/controllers/application_controller.rb

```diff
@@ -5,6 +5,148 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
+  end
+
+  get '/' do
+  	erb :index
+  end
+
+  get '/signup' do
+  	if logged_in?
+  		redirect to '/tweets'
+  	else
+  		erb :signup
+  	end
+  end
+
+  get '/login' do
+  	if logged_in?
+  		redirect to '/tweets'
+  	else
+  		erb :login
+  	end
+  end
+
+  post '/login' do
+  	user = User.find_by(username: params[:username])
+  	if user && user.authenticate(params[:password])
+  		session[:user_id] = user.id
+  		redirect to '/tweets'
+  	else
+  		redirect to '/signup'
+  	end
+  end
+
+  post '/signup' do
+  	if params[:username] == "" || params[:email] == "" || params[:password] == ""
+      redirect '/signup'
+    else
+    	user = User.new(:username => params[:username], :email => params[:email], :password => params[:password])
+    	user.save
+    	session[:user_id] = user.id
+  		redirect to '/tweets'
+  	end
+  end
+
+  get '/tweets' do
+  	if logged_in?
+  		@tweets = Tweet.all
+	  	erb :tweets
+	  else
+	  	redirect '/login'
+	  end
+  end
+
+  get '/tweets/new' do
+  	if logged_in?
+  		erb :'new_tweet'
+  	else
+  		redirect to '/login'
+  	end
+  end
+
+  post '/tweets' do
+    if params[:content] == ""
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.